File: src/decoders/partner-ard-wallet.ts
import type { PartnerARDWallet } from "~/models";

export const decodePartnerARDWallet = (wallet: any): PartnerARDWallet => ({
  name: wallet.libellePorteMonnaie,
  description: wallet.hintPorteMonnaie,
  warning: wallet.avecWarning,
  // Will turn `420,69 €` into `420.69`
  balance: parseInt(wallet.valeurSolde.replace(",", ".")),
  balanceDescription: wallet.hintSolde
});

File: library/src/commonMain/kotlin/ink/literate/pawnote/decoders/partner-ard-wallet.kt
package ink.literate.pawnote.decoders

import ink.literate.pawnote.models.PartnerARDWallet
import kotlinx.serialization.json.JsonObject
import kotlinx.serialization.json.boolean
import kotlinx.serialization.json.jsonPrimitive

fun decodePartnerARDWallet(wallet: JsonObject): PartnerARDWallet {
  return PartnerARDWallet(
      name = wallet["libellePorteMonnaie"]!!.jsonPrimitive.content,
      description = wallet["hintPorteMonnaie"]!!.jsonPrimitive.content,
      warning = wallet["avecWarning"]!!.jsonPrimitive.boolean,
      balance = wallet["valeurSolde"]!!.jsonPrimitive.content.replace(",", ".").toDouble(),
      balanceDescription = wallet["hintSolde"]!!.jsonPrimitive.content)
}
